apps/hash-graph/lib/graph/src/store/postgres/ontology/property_type.rs

```diff
@@ -35,87 +38,82 @@ impl<C: AsClient> PostgresStore<C> {
+                            new_graph_resolve_depths,
+                            temporal_axes.clone(),
+                        );
+                    }
+                }
+            }
+
+            if let Some(traversal_data) =
+                edges_to_traverse.get(&OntologyEdgeKind::ConstrainsValuesOn)
+            {
+                data_type_queue.extend(
+                    self.read_ontology_edges::<PropertyTypeVertexId, DataTypeVertexId>(
+                        traversal_data,
+                        ReferenceTable::PropertyTypeConstrainsValuesOn,
                     )
                     .await?
-                    {
-                        let data_type_vertex_id = data_type.vertex_id(time_axis);
+                    .map(|edge| {
+                        traversal_context.add_data_type_id(edge.right_endpoint.clone());
 
                         subgraph.insert_edge(
-                            &property_type_vertex_id,
+                            &edge.left_endpoint,
                             OntologyEdgeKind::ConstrainsValuesOn,
                             EdgeDirection::Outgoing,
-                            data_type_vertex_id.clone(),
+                            edge.right_endpoint.clone(),
                         );
 
-                        traversal_context.add_data_type_id(data_type_vertex_id.clone());
-
-                        data_type_queue.push((
-                            data_type_vertex_id,
-                            new_graph_resolve_depths,
-                            temporal_axes.clone()
-                        ));
-                    }
-                }
+                        (edge.right_endpoint, edge.resolve_depths, edge.temporal_axes)
+                    }),
+                );
+            }
 
-                if let Some(new_graph_resolve_depths) = graph_resolve_depths
-                    .decrement_depth_for_edge(
-                        OntologyEdgeKind::ConstrainsPropertiesOn,
-                        EdgeDirection::Outgoing,
-                    )
-                {
-                    for referenced_property_type in <Self as Read<PropertyTypeWithMetadata>>::read_vec(
-                        self,
-                        &Filter::<PropertyTypeWithMetadata>::for_ontology_edge_by_property_type_vertex_id(
-                            &property_type_vertex_id,
-                            OntologyEdgeKind::ConstrainsPropertiesOn,
-                            EdgeDirection::Incoming,
-                        ),
-                        Some(&temporal_axes),
+            if let Some(traversal_data) =
+                edges_to_traverse.get(&OntologyEdgeKind::ConstrainsPropertiesOn)
+            {
+                property_type_queue.extend(
+                    self.read_ontology_edges::<PropertyTypeVertexId, PropertyTypeVertexId>(
+                        traversal_data,
+                        ReferenceTable::PropertyTypeConstrainsValuesOn,
```

Comment:
Is this the wrong ReferenceTable?
```suggestion
                        ReferenceTable::PropertyTypeConstrainsPropertiesOn,
```

Reply:
Oh, good catch. I just noticed that we never use that edge kind in tests.